marketplace/google-analytics-app/src/Timeline.tsx

```diff
@@ -0,0 +1,164 @@
+import * as React from 'react';
+import { formatDate } from './utils';
+import styles from './styles';
+import {
+  SkeletonImage,
+  TextLink,
+  SkeletonContainer,
+  Paragraph
+} from '@contentful/forma-36-react-components';
+import { TimelineProps, TimelineState, ChartData } from './typings';
+
+const CHART_HEIGHT = 200;
+const externalUrlBase = 'https://analytics.google.com/analytics/web/#/report/content-pages';
+const externalUrlPageQuery = '_r.drilldown=analytics.pagePath:';
+
+export default class Timeline extends React.Component<TimelineProps, TimelineState> {
+  timeline!: HTMLElement;
+
+  onSuccess = ({ data }: { data: ChartData }) => {
+    this.setState({ loading: false });
+    this.props.onData(data);
+  };
+
+  onError = ({ error }: { error: Error }) => {
+    this.setState({ loading: false });
+    this.props.sdk.notifier.error(
+      `Google Analytics App couldn't get load your page view data (${error.message})`
+    );
+    this.props.onError();
+  };
+
+  constructor(props: TimelineProps) {
+    super(props);
+    this.state = {
+      timeline: undefined,
+      viewUrl: '',
+      loading: true
+    };
+  }
+
+  async componentDidMount() {
+    let viewUrl = '';
+    const { sdk, gapi } = this.props;
+
+    try {
+      const accounts = (await gapi.client.analytics.management.accountSummaries.list()) || [];
+      viewUrl = this.getExternalUrl(accounts);
+    } catch (e) {
+      const error = e.result ? e.result.error : e;
+      sdk.notifier.error(
+        `Google Analytics App couldn't get a link to your dashboard (${error.message})`
+      );
```

Comment:
Should we continue here or `return` early?

Reply:
The try block is only to get the url to the report in the google analytics website, so it's fine to continue without it. If we can't get the url, then we just don't render a link.